build.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+# A convenience script used to build our code. Takes one argument that specifies the workspace to build.
+
+# If sourcing file then return instead of exiting shell
```

Comment:
I don't think we need this much error handling, we should just be able to rely on this script running correctly.

That said, I would not be opposed to spawning a subshell to run everything first, and then just sourcing the `setup.bash` scripts after that subshell returns.